src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -812,6 +812,9 @@ protected virtual void Dispose(bool disposing)
                 _sender = new Ping();
 
                 timer?.Start();
+                // 'SendPingAsync' always uses the default synchronization context (threadpool).
```

Comment:
Default only because we add GetAwaiter().GetResult().

Reply:
No, it has nothing to do with `GetAwaiter().GetResult()`. `SendPingAsync` uses `ConfigureAwait(false)` on all `await` calls within it, that's why it always uses the default context.
Library APIs should always do that. When interacting with user code, meaning when user calls `await SendPingAsync(...)`, the user can decide if the continuation should carry on the current sync context or on the default (by add `.ConfigureAwait(false)`)

Reply:
Thanks! I see.